osquery/tables/events/darwin/openbsm_events.cpp

```diff
@@ -71,24 +81,52 @@ class OpenBSMSSHLoginSubscriber
                   const OpenBSMSubscriptionContextRef& sc);
 };
 
-REGISTER(OpenBSMExecVESubscriber, "event_subscriber", "process_events");
+REGISTER(OpenBSMProcEvSubscriber, "event_subscriber", "process_events");
 REGISTER(OpenBSMSSHLoginSubscriber, "event_subscriber", "user_events");
 
-void OpenBSMExecVESubscriber::configure() {
+void OpenBSMProcEvSubscriber::configure() {
   std::vector<size_t> event_ids{
-      23, // execve
-      43190, // AUX_POSIX_SPAWN
+      AUE_EXECVE,
+      AUE_POSIX_SPAWN,
+      AUE_FORK,
+      AUE_VFORK,
+      AUE_FORK1,
+      AUE_DARWIN_RFORK,
+      AUE_RFORK,
+      AUE_EXIT,
   };
   for (const auto& evid : event_ids) {
     auto sc = createSubscriptionContext();
     sc->event_id = evid;
-    subscribe(&OpenBSMExecVESubscriber::Callback, sc);
+    subscribe(&OpenBSMProcEvSubscriber::Callback, sc);
   }
 }
 
-Status OpenBSMExecVESubscriber::Callback(
+Status OpenBSMProcEvSubscriber::Callback(
     const OpenBSMEventContextRef& ec, const OpenBSMSubscriptionContextRef& sc) {
+  switch (sc->event_id) {
+  case AUE_EXECVE:
+  case AUE_POSIX_SPAWN:
+    return handleExec(ec);
+
+  case AUE_FORK:
+  case AUE_VFORK:
+  case AUE_FORK1:
+  case AUE_DARWIN_RFORK:
+  case AUE_RFORK:
+    return handleFork(ec);
+
+  case AUE_EXIT:
+    return handleExit(ec);
+  }
+
+  return Status(1, "Unexpected event");
+}
+
+Status OpenBSMProcEvSubscriber::handleExec(const OpenBSMEventContextRef& ec) {
   Row r;
+  uint32_t pid;
```

Comment:
Though it should probably never happen that we get through all tokens without finding a pid, we should initialize this so we don't get undefined behaviour as a possibility on line 200.

Reply:
Then I think we're good to merge.

Reply:
Good catch, fixed!